dev/TreeView/TreeViewList.cpp

```diff
@@ -320,6 +320,14 @@ void TreeViewList::PrepareContainerForItemOverride(winrt::DependencyObject const
     {
         bool hasChildren = itemContainer.HasUnrealizedChildren() || itemNode->HasChildren();
         itemContainer.GlyphOpacity(hasChildren ? 1.0 : 0.0);
+        if (itemContainer.IsExpanded() != itemNode->IsExpanded()) {
+            const DispatcherHelper dispatcher{ *this };
```

Comment:
I'm curious why you need to run this though the dispatcher async.

Reply:
> I'm curious why you need to run this though the dispatcher async.
> 
It crashes if we don't use a dispatcher, I think it's still the same reason, changing expand state will change ListView's ItemsSource as we are adding/removing items from ListView, and this can't be done during layout.

> Also, should'nt it be the other way ? we are creating the container based on the data, so I would think we need to set the container's value to what the node has.
>
Yeah this is confusing, in content mode, TreeViewItem (which is the container) holds the truth since we are using data binding, and in node mode it's the opposite, TreeViewNode holds the truth (at line 334 we are updating container based on TreeViewNode's state).